File: app/src/main/java/com/gooner10/popularmoviesapp/moviemain/MovieFragment.java
package com.gooner10.popularmoviesapp.moviemain;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.gooner10.popularmoviesapp.R;
import com.gooner10.popularmoviesapp.data.MovieItem;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import hugo.weaving.DebugLog;

/**
 * Movie Fragment to display the Movie List in GridLayout
 */

public class MovieFragment extends Fragment implements MovieContract.View {
    private final String TAG = "MovieFragment";
    private MovieFragmentAdapter movieFragmentAdapter;
˚
    @BindView(R.id.recyclerViewMovie)
    RecyclerView movierecyclerview;

    public MovieFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NotNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_movie, container, false);

        // Bind all of the view
        ButterKnife.bind(this, view);

        movierecyclerview.setLayoutManager(new GridLayoutManager(getActivity(), 2));
        movieFragmentAdapter = new MovieFragmentAdapter(getActivity(), new ArrayList<MovieItem>());
        movierecyclerview.setAdapter(movieFragmentAdapter);
        Log.i(TAG, "onCreateView movieFragmentAdapter: " + movieFragmentAdapter);
        MovieViewModel movieViewModel = ViewModelProviders.of(getActivity()).get(MovieViewModel.class);
        Log.i(TAG, "loadData getMovieDataList: " + movieViewModel.getMovieDataList());
        movieViewModel.getMovieDataList().observe(getActivity(), new Observer<List<MovieItem>>() {
            @Override
            @DebugLog
            public void onChanged(@Nullable List<MovieItem> movieDataList) {
                displayMovieList(movieDataList);
            }
        });
        return view;
    }

    @Override
    @DebugLog
    public void displayMovieList(List<MovieItem> movieDataList) {
        Log.i(TAG, "displayMovieList movieFragmentAdapter: " + movieFragmentAdapter);
        movieFragmentAdapter.setData(movieDataList);
    }

}

File: app/src/main/java/com/gooner10/popularmoviesapp/moviemain/MovieFragment.kt
package com.gooner10.popularmoviesapp.moviemain

import android.content.Context
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.GridLayoutManager
import com.gooner10.popularmoviesapp.R
import com.gooner10.popularmoviesapp.data.MovieItem
import hugo.weaving.DebugLog
import kotlinx.android.synthetic.main.fragment_movie.view.*
import java.util.*

/**
 * Movie Fragment to display the Movie List in GridLayout
 */

class MovieFragment : Fragment(), MovieContract.View {
    companion object {
        private val TAG = "MovieFragment"
    }

    private var movieFragmentAdapter: MovieFragmentAdapter? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        val view = inflater.inflate(R.layout.fragment_movie, container, false)

        view.recyclerViewMovie.layoutManager = GridLayoutManager(activity, 2)
        movieFragmentAdapter = MovieFragmentAdapter(activity as Context, ArrayList())
        view.recyclerViewMovie.adapter = movieFragmentAdapter
        Log.i(TAG, "onCreateView movieFragmentAdapter: " + movieFragmentAdapter!!)
        val movieViewModel = ViewModelProviders.of(activity!!).get(MovieViewModel::class.java)
        Log.i(TAG, "loadData getMovieDataList: " + movieViewModel.movieDataList)
        movieViewModel.movieDataList.observe(activity!!, Observer { movieDataList -> displayMovieList(movieDataList) })
        return view
    }

    @DebugLog
    override fun displayMovieList(movieDataList: List<MovieItem>?) {
        Log.i(TAG, "displayMovieList movieFragmentAdapter: " + movieFragmentAdapter!!)
        movieFragmentAdapter!!.setData(movieDataList)
    }

}
